ark/type/parser/objectLiteral.ts

```diff
@@ -24,8 +28,41 @@ const stringAndSymbolicEntriesOf = (o: Record<string | symbol, unknown>) => [
 export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
 	const required: Inner<"required">[] = []
 	const optional: Inner<"optional">[] = []
+
+	// We only allow a spread operator to be used as the first key in an object
+	// because to match JS behavior any keys before the spread are overwritten
+	// by the values in the target object, so there'd be no useful purpose in having it
+	// anywhere except for the beginning.
+	// Discussion in ArkType Discord:
+	// https://discord.com/channels/957797212103016458/1103023445035462678/1182814502471860334
+	let hasSeenFirstKey = false
+
 	for (const entry of stringAndSymbolicEntriesOf(def)) {
 		const result = parseEntry(entry)
+
+		if (result.kind === "spread" && hasSeenFirstKey) {
```

Comment:
I'd write a single if statement here and have the error case be a nested if condition at the beginning of it so you can get rid of the else.